packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -903,4 +964,74 @@ contract OPContractsManager is ISemver {
         if (msg.sender != upgradeController) revert OnlyUpgradeController();
         isRC = _isRC;
     }
+
+    function getGameConstructorParams(IFaultDisputeGame _disputeGame)
+        internal
+        view
+        returns (IFaultDisputeGame.GameConstructorParams memory)
+    {
+        IFaultDisputeGame.GameConstructorParams memory params = IFaultDisputeGame.GameConstructorParams({
+            gameType: _disputeGame.gameType(),
+            absolutePrestate: _disputeGame.absolutePrestate(),
+            maxGameDepth: _disputeGame.maxGameDepth(),
+            splitDepth: _disputeGame.splitDepth(),
+            clockExtension: _disputeGame.clockExtension(),
+            maxClockDuration: _disputeGame.maxClockDuration(),
+            vm: _disputeGame.vm(),
+            weth: _disputeGame.weth(),
+            anchorStateRegistry: _disputeGame.anchorStateRegistry(),
+            l2ChainId: _disputeGame.l2ChainId()
+        });
+        return params;
+    }
+
+    /// @notice For a given game type, does the following:
+    /// 1. Get and upgrade the DelayedWETH Proxy
+    /// 2. Deploy a new game
+    /// 3. Set the new game as the implementation on the OptimismPortal
+    function deployAndSetNewGameImpl(
+        IProxyAdmin _proxyAdmin,
+        IDisputeGame _currentGame,
+        IAnchorStateRegistry _newAnchorStateRegistryProxy,
+        GameType _gameType,
+        Blueprints memory _blueprints,
+        Implementations memory _implementations,
+        ISystemConfig.Addresses memory _opChainAddrs,
+        uint256 _l2ChainId
+    )
```

Comment:
I really dislike how many args this has. A few values could be obtained through another external call, but I felt it was better to pass them in here:
- `_implementations`
- `_l2ChainId`

Also, `_blueprints` is only used in this function, so we could make the call to `thisOPCM.blueprints()` in this function, but I prefer to keep the external calls to `thisOPCM` closer together at the top of the function as a kind of initialization step.